client/framework/qb.lua

```diff
@@ -17,7 +17,10 @@ local playerItems = setmetatable({}, {
 })
 
 local function setPlayerItems()
-    playerItems = {}
+    for k, _ in pairs(playerItems) do
+        playerItems[k] = nil
+    end
+    
```

Comment:
`table.wipe(playerItems)`?

Reply:
Updated to use table.wipe.

I assume you meant QBCore:Player:SetPlayerData
However would we want to do this table wipe if the player is using ox_inventory?

Reply:
Does qb not support logout at all? That's when I'd assume you want to wipe the table.
No need to wipe for ox_inventory unless logout is a thing (which is why I suggested OnPlayerLoaded).

Forgot there's no events for individual data changes (like just items, or a single item) so it's kinda gross in that file.. setting up item count everytime something changes.

Reply:
I'm not seeing how this is related to login/logout. The bug we're trying to solve is that when item data changes, say from the user dropping an item, the corresponding entry within playerItems needs to be cleared so that the target code has an updated understanding of the player's items. QBCore:Player:SetPlayerData currently is triggered every time a player's items changes; and since the table is rebuilt within setPlayerItems, the missing step is wiping the table beforehand so that ghost items don't linger around.